jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/GroovyMetricEnvironment.java

```diff
@@ -219,19 +227,43 @@ public LongHistogram getLongHistogram(
    * @param description metric description
    * @param unit - metric unit
    * @param updater - the value updater
+   * @return the ObservableDoubleMeasurement for the gauge
    */
-  public void registerDoubleValueCallback(
+  public ObservableDoubleMeasurement registerDoubleValueCallback(
       final String name,
       final String description,
       final String unit,
       final Consumer<ObservableDoubleMeasurement> updater) {
-    meter
-        .gaugeBuilder(name)
-        .setDescription(description)
-        .setUnit(unit)
-        .buildWithCallback(
-            proxiedDoubleObserver(
-                name, description, unit, InstrumentType.OBSERVABLE_GAUGE, updater));
+    int descriptorHash =
+        InstrumentDescriptor.create(
+                name,
+                description,
+                unit,
+                InstrumentType.OBSERVABLE_GAUGE,
+                InstrumentValueType.DOUBLE)
+            .hashCode();
+
+    // Only build the instrument if it isn't already in the registry
+    ObservableMeasurement obs = instrumentOnceRegistry.get(descriptorHash);
+    if (obs == null) {
+      obs = meter.gaugeBuilder(name).setDescription(description).setUnit(unit).buildObserver();
+      instrumentOnceRegistry.put(descriptorHash, obs);
+      // If an updater was not provided, the measurement is expected to be added
```

Comment:
Does this describe an intentional case or is it for safety?

Reply:
This is an intentional case, providing for the instruments for composite data to be grouped into the same batchCallback.